### PR TITLE
Add more tests and S gate

### PR DESCRIPTION
**Pull request by VarLad**
Added S gate translation and test for KronBlock

### CODE REVIEW

File: src/YaoPastaQ.jl [hunk 1]
```diff
@@ -26,7 +26,8 @@ genlist!(plist, ::YGate, locs, controls) = push!(plist, ("Y", locs[1]))
 genlist!(plist, ::HGate, locs, controls) = push!(plist, ("H", locs[1]))
 genlist!(plist, ::ZGate, locs, controls) = push!(plist, ("Z", locs[1]))
 genlist!(plist, ::Scale{Val{im}, 1, YGate}, locs, controls) = push!(plist, ("iY", locs[1]))
-genlist!(plist, ::TGate, locs, controls) = push!(plist, ("π/8", locs[1])) 
+genlist!(plist, ::TGate, locs, controls) = push!(plist, ("π/8", locs[1]))
+genlist!(plist, ::ConstGate.SGate, locs, controls) = push!(plist, ("Phase", locs[1]))
```
Comment (Roger-luo) [inline src/YaoPastaQ.jl:30]: maybe you want to just write `using Yao.ConstGate` on top of this module

Comment (Roger-luo) [reply src/YaoPastaQ.jl:30]: if you want the `SGate` symbol, but writing the full path is also fine

**Review by Roger-luo — APPROVED** [review]
just FYI, we are switching to coprac, which means I will only approve PRs for those who has merge rights, and you can merge the PR yourself once it's approved. Just remember to do a `squash merge` instead of a `merge` to clean up the commit messages

**Comment by VarLad** [comment]
@Roger-luo Does this PR require anything else?
If not, I think its good to merge

**Pull request merged**